src/System.Data.SqlClient/tests/ManualTests/SQL/SqlFileStreamTest/SqlFileStreamTest.cs

```diff
@@ -0,0 +1,217 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Data.SqlTypes;
+using System.IO;
+using Xunit;
+
+namespace System.Data.SqlClient.ManualTesting.Tests
+{
+    public static class SqlFileStreamTest
+    {
+        private static bool IsFileStreamEnvironmentSet() => DataTestUtility.IsFileStreamSetup();
+        private static bool AreConnectionStringsSetup() => DataTestUtility.AreConnStringsSetup();
+        private static bool IsIntegratedSecurityEnvironmentSet() => DataTestUtility.IsIntegratedSecuritySetup();
+
+        private static int[] s_insertedValues = { 11 , 22 };
+
+        [PlatformSpecific(TestPlatforms.Windows)]
+        [ConditionalFact(nameof(IsFileStreamEnvironmentSet), nameof(IsIntegratedSecurityEnvironmentSet), nameof(AreConnectionStringsSetup))]
+        public static void ReadFilestream()
+        {
+            using (SqlConnection connection = new SqlConnection(DataTestUtility.TcpConnStr))
+            {
+                connection.Open();
+                string tempTable = SetupTable(connection);
+                int nRow = 0;
+                byte[] retrievedValue;
+                SqlCommand command = new SqlCommand($"SELECT Photo.PathName(), GET_FILESTREAM_TRANSACTION_CONTEXT(),EmployeeId FROM {tempTable} ORDER BY EmployeeId", connection);
+
+                SqlTransaction transaction = connection.BeginTransaction(IsolationLevel.ReadCommitted);
+                command.Transaction = transaction;
+
+                using (SqlDataReader reader = command.ExecuteReader())
+                {
+                    while (reader.Read())
+                    {
+                        // Get the pointer for the file.
+                        string path = reader.GetString(0);
+                        byte[] transactionContext = reader.GetSqlBytes(1).Buffer;
+                        
+                        // Create the SqlFileStream  
+                        using (Stream fileStream = new SqlFileStream(path, transactionContext, FileAccess.Read, FileOptions.SequentialScan, allocationSize: 0))
```

Comment:
Explained in the comment above and more information in comments #15652
Also, this is Windows-only because SQL Server on Linux doesn't support `FileStream` currently.
In case of Linux Client and Windows SQL Server, as per discussions with Robert Dorr, currently there is no way to open a file over SMB and pass the Extended Attributes on Linux.